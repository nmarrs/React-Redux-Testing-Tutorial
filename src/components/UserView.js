import React from 'react'
import PropTypes from 'prop-types'

import TaskColumn from './TaskColumn'

const UserView = (props) => {
  const {
    toDoTasks,
    inProgressTasks,
    finishedTasks,
    toggleEditTaskModal,
    editTaskModalOpen,
    isAdminView,
    editTask,
    currentTask
  } = props

  return (
    <div>
      <h3 className='mt-3'><i className="fas fa-tasks"></i> Current User Tasks</h3>
      <div className='container'>
        <div className='row'>
          <TaskColumn
            title={'To Do'}
            tasks={toDoTasks}
            toggleEditTaskModal={toggleEditTaskModal}
            editTaskModalOpen={editTaskModalOpen}
            isAdminView={isAdminView}
            editTask={editTask}
            currentTask={currentTask} />
          <TaskColumn
            title={'In Progress'}
            tasks={inProgressTasks}
            toggleEditTaskModal={toggleEditTaskModal}
            editTaskModalOpen={editTaskModalOpen}
            isAdminView={isAdminView}
            editTask={editTask}
            currentTask={currentTask} />
          <TaskColumn
            title={'Finished'}
            tasks={finishedTasks}
            toggleEditTaskModal={toggleEditTaskModal}
            editTaskModalOpen={editTaskModalOpen}
            isAdminView={isAdminView}
            editTask={editTask}
            currentTask={currentTask} />
        </div>
      </div>
    </div>
  )
}

UserView.propTypes = {
  toDoTasks: PropTypes.array.isRequired,
  inProgressTasks: PropTypes.array.isRequired,
  finishedTasks: PropTypes.array.isRequired,
  isAdminView: PropTypes.bool.isRequired,
  editTaskModalOpen: PropTypes.bool.isRequired,
  toggleEditTaskModal: PropTypes.func.isRequired,
  currentTask: PropTypes.object
}

export default UserView
