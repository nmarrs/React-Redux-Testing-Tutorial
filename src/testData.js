export let testTask = {
  id: 1,
  title: 'Unload merchandise',
  priority: 3,
  timeEstimate: 2,
  notes: 'Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 150
}

export let testTask1 = {
  id: 2,
  title: 'Stock shelves with unpacked items',
  priority: 1,
  timeEstimate: 10,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 10
}

export let testTask2 = {
  id: 3,
  title: 'Stock shelves with unpacked items',
  priority: 5,
  timeEstimate: 2,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 250
}
