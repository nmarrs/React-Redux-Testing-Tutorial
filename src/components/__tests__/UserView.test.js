import React from 'react'
import { shallow } from 'enzyme'

import UserView from '.././UserView'
import { testTask, testTask1, testTask2 } from '../../testData'

describe('UserView', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<UserView
      toDoTasks={[ testTask ]}
      inProgressTasks={[ testTask1 ]}
      finishedTasks={[ testTask2 ]}
      isAdminView={true}
      editTaskModalOpen={false}
      toggleEditTaskModal={jest.fn()} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
