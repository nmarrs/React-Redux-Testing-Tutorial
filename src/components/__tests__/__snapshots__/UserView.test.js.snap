// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders correctly 1`] = `
<div>
  <h3
    className="mt-3"
  >
    <i
      className="fas fa-tasks"
    />
     Current User Tasks
  </h3>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <TaskColumn
        editTaskModalOpen={false}
        isAdminView={true}
        tasks={
          Array [
            Object {
              "completionTime": "",
              "currentStatus": "To Do",
              "endTime": "",
              "feedback": "",
              "id": 1,
              "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
              "priority": 3,
              "rankWeight": 150,
              "startTime": "",
              "timeEstimate": 2,
              "title": "Unload merchandise",
            },
          ]
        }
        title="To Do"
        toggleEditTaskModal={[MockFunction]}
      />
      <TaskColumn
        editTaskModalOpen={false}
        isAdminView={true}
        tasks={
          Array [
            Object {
              "completionTime": "",
              "currentStatus": "To Do",
              "endTime": "",
              "feedback": "",
              "id": 2,
              "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
              "priority": 1,
              "rankWeight": 10,
              "startTime": "",
              "timeEstimate": 10,
              "title": "Stock shelves with unpacked items",
            },
          ]
        }
        title="In Progress"
        toggleEditTaskModal={[MockFunction]}
      />
      <TaskColumn
        editTaskModalOpen={false}
        isAdminView={true}
        tasks={
          Array [
            Object {
              "completionTime": "",
              "currentStatus": "To Do",
              "endTime": "",
              "feedback": "",
              "id": 3,
              "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
              "priority": 5,
              "rankWeight": 250,
              "startTime": "",
              "timeEstimate": 2,
              "title": "Stock shelves with unpacked items",
            },
          ]
        }
        title="Finished"
        toggleEditTaskModal={[MockFunction]}
      />
    </div>
  </div>
</div>
`;
